tests/unit/libbpf_test.cpp

```diff
@@ -3388,34 +3393,59 @@ _test_batch_iteration_maps(fd_t& map_fd, uint32_t batch_size, bpf_map_batch_opts
         REQUIRE(result == 0);
         // Number of entries retrieved (batch_size_count) should be less than or equal to requested_batch_size.
         REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_keys.size()));
-        REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_values.size()));
+        REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_values.size()) / value_size);
 
         batch_keys.resize(batch_size_count);
-        batch_values.resize(batch_size_count);
+        batch_values.resize(batch_size_count * value_size);
         returned_keys.insert(returned_keys.end(), batch_keys.begin(), batch_keys.end());
         returned_values.insert(returned_values.end(), batch_values.begin(), batch_values.end());
 
         in_batch = &out_batch;
     }
 
     REQUIRE(returned_keys.size() == batch_size);
-    REQUIRE(returned_values.size() == batch_size);
-
-    std::sort(returned_keys.begin(), returned_keys.end());
-    std::sort(returned_values.begin(), returned_values.end());
+    REQUIRE(returned_values.size() == batch_size * value_size);
 
     // Verify the returned keys and values.
+    uint32_t key = 0;
+    for (uint32_t i = 0; i < batch_size; i++) {
+        key = returned_keys[i];
+        for (int cpu = 0; cpu < num_of_cpus; cpu++) {
+            REQUIRE(returned_values[(i * value_size) + cpu] == static_cast<uint64_t>(key) * 2ul);
+        }
+    }
+    // Hash maps do not guarantee order of keys.
+    // The keys in the returned_keys vector should be in the range [0, batch_size-1].
+    std::sort(returned_keys.begin(), returned_keys.end());
     for (uint32_t i = 0; i < batch_size; i++) {
         REQUIRE(returned_keys[i] == i);
-        REQUIRE(returned_values[i] == static_cast<uint64_t>(i) * 2ul);
     }
 }
 
 void
-_test_maps_batch(bpf_map_type map_type)
+_test_maps_batch(bpf_map_type map_type, bool percpu)
```

Comment:
It looks like percpu can be inferred from the map_type. Is that correct? If so, perhaps it need not be an input parameter and can just be determined based on the map type.

Reply:
Re-opening this. The other code change in your PR uses BPF_MAP_TYPE_PER_CPU() macro. Is there a reason this cannot be used?